petab/core.py

```diff
@@ -220,16 +220,17 @@ def assignment_rules_to_dict(
     """
     result = {}
 
-    # iterate over all parameters
-    for p in sbml_model.getListOfParameters():
-        parameter_id = p.getId()
+    # iterate over rules
+    for rule in sbml_model.getListOfRules():
+        if rule.getTypeCode() != libsbml.SBML_ASSIGNMENT_RULE:
```

Comment:
what cases does this exclude e.g.? if you ran into an error without this, would be good to document

Reply:
It allows you to have parameters named `observable_*`, `sigma_*` independently of whether they are used as targets of assignment rules. It is pretty much documented in the markdown document. The code just did something different...